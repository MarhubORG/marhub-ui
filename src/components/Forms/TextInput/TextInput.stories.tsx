import * as React from 'react';
import { storiesOf } from '@storybook/react';
import { ThemeProvider } from 'styled-components';
import { mainTheme } from '../../../styles/themes';
import TextInput from './TextInput';

// storiesOf('TextInput', module).add('TextInput Component', () => (
//   <ThemeProvider theme={mainTheme}>
//     <TextInput
//       htmlFor="organization"
//       labelText="Organization:"
//       placeholder="Organization"
//       name="organization"
//       value=""
//       onChange={(): void => {}}
//     />
//   </ThemeProvider>
// ));
