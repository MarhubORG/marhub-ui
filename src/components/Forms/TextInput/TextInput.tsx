import React from 'react';
import styled from 'styled-components';

interface Props {
  htmlFor: string;
  labelText: string;
  placeholder: string;
  name: string;
  value: string;
  onChange(message: string): void;
  password?: boolean;
}

function TextInput(props: Props): JSX.Element {
  return (
    <div>
      <Label htmlFor={props.htmlFor}>{props.labelText}</Label>
      <TextInputHTML
        type={props.password ? 'password' : 'text'}
        placeholder={props.placeholder}
        name={props.name}
        value={props.value}
        onChange={(e): void => props.onChange(e.target.value)}
      />
    </div>
  );
}

const TextInputHTML = styled.input`
  height: 2rem;
  min-width: 18rem;
  max-width: 18rem;
  border-radius: 0.2rem;
  font-size: 0.9rem;
  padding-left: 1rem;
  padding-right: 1rem;
  border: 1px solid #edf1f7;
`;

const Label = styled.label`
  display: block;
  color: ${({ theme }): string => theme.grayText};
  font-family: Open Sans, sans-serif;
  font-size: 0.75rem;
  font-weight: 700;
  line-height: 1rem;
  margin: 0.5rem 0rem;
`;

export default TextInput;
