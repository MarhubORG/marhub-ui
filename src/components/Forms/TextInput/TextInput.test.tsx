import React from 'react';
import renderer from 'react-test-renderer';
import TextInput from './TextInput';

it('renders correctly', () => {
  const tree = renderer
    .create(
      <TextInput
        htmlFor="organization"
        labelText="Organization:"
        placeholder="Organization"
        name="organization"
        value="val"
        onChange={jest.fn()}
      />
    )
    .toJSON();
  expect(tree).toMatchSnapshot();
});
