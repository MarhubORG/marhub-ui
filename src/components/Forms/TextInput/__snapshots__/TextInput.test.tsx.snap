// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
.c1 {
  height: 2rem;
  min-width: 18rem;
  max-width: 18rem;
  border-radius: 0.2rem;
  font-size: 0.9rem;
  padding-left: 1rem;
  padding-right: 1rem;
  border: 1px solid #edf1f7;
}

.c0 {
  display: block;
  font-family: Open Sans,sans-serif;
  font-size: 0.75rem;
  font-weight: 700;
  line-height: 1rem;
  margin: 0.5rem 0rem;
}

<div>
  <label
    className="c0"
    htmlFor="organization"
  >
    Organization:
  </label>
  <input
    className="c1"
    name="organization"
    onChange={[Function]}
    onKeyDown={[Function]}
    placeholder="Organization"
    type="text"
    value="val"
  />
</div>
`;
